internal/provider/resource_cloudflare_account.go

```diff
@@ -0,0 +1,150 @@
+		UpdateContext: resourceCloudflareAccountUpdate,
+		DeleteContext: resourceCloudflareAccountDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Description: heredoc.Doc(`
+			Provides a Cloudflare Account resource. Account is the basic resource for
+			working with Cloudflare zones, teams and users. Requires the tenant entitlement
+		`),
+	}
+}
+
+func resourceCloudflareAccountCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+	accountName := d.Get("name").(string)
+	var accountType string
+	if d.Get("type").(string) == "" {
+		accountType = accountTypeStandard
+	} else if d.Get("type").(string) == accountTypeStandard || d.Get("type").(string) == accountTypeEnterprise {
+		accountType = d.Get("type").(string)
+	} else {
+		return diag.FromErr(fmt.Errorf("invalid account type %s", d.Get("type").(string)))
+	}
+
+	tflog.Info(ctx, fmt.Sprintf("Creating Cloudflare Account: name %s", accountName))
+
+	account := cloudflare.Account{
+		Name: accountName,
+		Type: accountType,
+	}
+	acc, err := client.CreateAccount(ctx, account)
+
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error creating account %q: %w", accountName, err))
+	}
+
+	d.SetId(acc.ID)
+
+	return resourceCloudflareAccountRead(ctx, d, meta)
+}
+
+func getCloudflareAccontFromId(accountID string, client *cloudflare.API, ctx context.Context) (cloudflare.Account, error) {
+	accs, _, err := client.Accounts(ctx, cloudflare.AccountsListParams{})
+	if err != nil {
+		return cloudflare.Account{}, err
+	} else {
+		for _, acc := range accs {
+			if acc.ID == accountID {
+				return acc, nil
+			}
+		}
+
+		return cloudflare.Account{}, fmt.Errorf("Account %s does not exist", accountID)
+	}
+}
+
+func resourceCloudflareAccountRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+	accountID := d.Id()
+
+	foundAcc, err := getCloudflareAccontFromId(accountID, client, ctx)
+	tflog.Debug(ctx, fmt.Sprintf("AccountDetails error: %#v", err))
+
+	if err != nil || foundAcc.ID == "" {
+		var notFoundError *cloudflare.NotFoundError
+		if errors.As(err, &notFoundError) {
+			tflog.Info(ctx, fmt.Sprintf("Account %s no longer exists", d.Id()))
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(fmt.Errorf("error finding Account %q: %w", d.Id(), err))
+	}
+
+	tflog.Debug(ctx, fmt.Sprintf("AccountDetails: %#v", foundAcc))
+
+	d.Set("name", foundAcc.Name)
+	d.Set("type", foundAcc.Type)
+	d.Set("enforce_twofactor", foundAcc.Settings.EnforceTwoFactor)
+
+	return nil
+}
+
+func resourceCloudflareAccountUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+	accountID := d.Id()
+	foundAcc, err := getCloudflareAccontFromId(accountID, client, ctx)
+
+	tflog.Debug(ctx, fmt.Sprintf("AccountDetails error: %#v", err))
```

Comment:
similar to https://github.com/cloudflare/terraform-provider-cloudflare/pull/1902/files#r967906187. this should be full error handling.

Reply:
just cleaned this useless remaining debug and added the error management for it.